### PR TITLE
LPS-168177 Allow to hide actions in browser tag

### PR DESCRIPTION
**Pull request by adolfopa**
# What is this about?

https://issues.liferay.com/issues/browse/LPS-168177

Add an attribute to the tag that allows to hide unnecessary actions.

The new `actions` attribute accepts a comma separated string with components:

- "all"
- "add-folder"
- "delete"
- "none"
- "rename"
- "upload"

Leaving the attribute empty, or passing "all" is equivalent to passing all available options. Passing "none" will remove all actions. Individual actions can be specified separated by commas (e.g. "add-folder,delete,upload").

image

# How do I test it?

To fully test it, you'll need to manually change the "view_images.jsp" file in blogs-web. Checking that blogs only displays the "delete" action may be enough if you're confident.

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by WYZ-Issac** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
Rebased and added an extra baseline commit.

**Comment by adolfopa** [comment]
ci:reevaluate:1287401_5201

**Comment by adolfopa** [comment]
ci:test:relevant

File: modules/apps/document-library/document-library-taglib/src/main/java/com/liferay/document/library/taglib/servlet/taglib/RepositoryBrowserTag.java [hunk 1]
```diff
@@ -105,6 +120,26 @@ _folderModelResourcePermission, _getFolderId(),
...
+
+		Set<String> set = SetUtil.fromArray(actions.split(",\\s*"));
+
+		if (actions.contains("none")) {
+			return Collections.emptySet();
+		}
+
+		if (actions.contains("all")) {
+			return _allActions;
+		}
```
[the remark this answers is not in the document]
Comment (marcogalluzzi) [reply modules/apps/document-library/document-library-taglib/src/main/java/com/liferay/document/library/taglib/servlet/taglib/RepositoryBrowserTag.java:138]: I would also rename variable _set_ to _actionsSet_ or similar.

**Comment by adolfopa** [comment]
@marcogalluzzi Updated code based on your suggestions. I've kept the constants inlined; they're internal to the module and they're mostly symmetrical to js action names (which are inlined as well).

**Comment by adolfopa** [comment]
ci:test:relevant

File: modules/apps/document-library/document-library-taglib/src/main/java/com/liferay/document/library/taglib/servlet/taglib/RepositoryBrowserTag.java [hunk 2]
```diff
@@ -120,24 +120,24 @@ _folderModelResourcePermission, _getFolderId(),
...
-	private Set<String> _getActions() {
+	private Set<String> _getActionsSet() {
 		if (Validator.isBlank(getActions())) {
 			return _allActions;
 		}
 
 		String actions = getActions();
 
-		Set<String> set = SetUtil.fromArray(actions.split(",\\s*"));
+		Set<String> actionsSet = SetUtil.fromArray(actions.split("\\s*,\\s*"));
```
Comment (marcogalluzzi) [inline modules/apps/document-library/document-library-taglib/src/main/java/com/liferay/document/library/taglib/servlet/taglib/RepositoryBrowserTag.java:130]: This regular expression doesn't take into account actions at the start or the end of the actions string if string starts or ends with spaces.

**Review by marcogalluzzi — CHANGES_REQUESTED** [review]

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**